adazzle/react-data-grid
# Set up storybook deployment workflow

- Split CI into two jobs:
  - `test` for linting/typechecking/jesting
    - building the storybook will now be done in the other job 👇 
  - `storybook` for building the storybook, and to automatically deploy it
    - PRs will only build the storybook, no deployment
- on `canary` branch pushes
  - checkout `gh-pages`
  - remove the existing `canary` directory
  - rename `storybook-static` to `canary`
  - stage changes, commit and push
- on `master` branch pushes
  - should create a clean/orphan `gh-pages` branch
    - not sure git won't throw an error, we'll see
  - stage the files under the `storybook-static` directory, then move those files to the repo root
  - restores the `canary` directory in the "original" `gh-pages` branch
  - commit and push

https://adazzle.github.io/react-data-grid/canary/
